server/pulp/plugins/distributor.py

```diff
@@ -337,3 +355,20 @@ def cancel_publish_group(self, call_request, call_report):
         :raise SystemExit: raised through a call to sys.exit()
         """
         sys.exit()
+
+    @staticmethod
+    def ensure_all_units_downloaded(repo_group):
+        """
+        Checks the database to make sure all units in the repos have been downloaded. If not, raises
+        an exception.
+
+        :param repo_group:  RepositoryGroup instance that should be checked
+        :type  repo_group:  pulp.plugins.model.RepositoryGroup
+
+        :raises exceptions.PulpCodedException:  if any unit in the group has un-downloaded files
+        """
+        bad_repo_ids = filter(lambda x: not repo_controller.has_all_units_downloaded(x),
+                              repo_group.repo_ids or [])
```

Comment:
I also looked for some usage to find non list instances, and I couldn't find any. If there are usages that are non list, this statement would mask that problem in our code even further. Generally I think experiencing a problem and fixing it is better than maintaining code that deals with inconsistent data types. Besides pulp-smash should find it before our users do.